### DISCUSSION TITLE
Subscriptions & SignalR

### DISCUSSION THREAD

**Discussion by ItsRobbAllen** [ravendb/ravendb#16060] [Q&A]
Not exactly a _Raven_ question, per se, but is anyone doing SignalR messages inside of a Subscription?

I have microservices that initiate subscriptions when the service starts. One of these subscriptions watches for specific document changes & I want to notify the UI that something happened via WebSockets. 

I build the subscriptions in Program.cs as singletons and start them up. I  use DI to inject the IHubContext, but when the subscriptions run, the Hub always shows 0 connections and no messages ever are received by the UI client.

Not sure what I'm missing why the hub context isn't pointing to the actual context, just wanted to see if anyone here might be doing something similar.

**Comment by georgiosd** [comment]
We've done this with the Changes API, never with the subscriptions API, although we use it heavily elsewhere.

Your description sounds fine so I don't think I could help without a minimal repro.

There are the debugging steps I would take:
- Make sure you can connect to the SignalR hub before anything - this has only to do with your SignalR setup, not the integration code.
- Make sure the subscription is actually executed.
- Mare sure a message is posted when the subscription runs.

**Reply by ItsRobbAllen** [reply]
Everything else in the system works fine. I can get WS messages back & forth, just not out of the subscription. That said, the changes API looks like a better fit for what I'm wanting to do. Just having problems with it. I've never worked with it and the sample code on the documentation page doesn't work, so I'm working through that.

**Comment by ayende** [comment]
Are you using the subscription solely to feed SignalR? If so, you don't need any reliability, do you?

**Reply by ItsRobbAllen** [reply]
I was going to say no, then I realized, that's pretty much the end result. I was thinking I was going to do some other things with the subscription, but now I realize that's not the case. I think the Change API will work better for me in this case, however I've never worked with it and the sample code throws an error.

Going to try to figure it out on my own for a bit before asking here.
